runtime/reconcilers/canvas.go

```diff
@@ -122,3 +129,77 @@ func (r *CanvasReconciler) checkAnyComponentHasValidSpec(ctx context.Context, re
 	}
 	return false, nil
 }
+
+// validateMetricsViewTimeConsistency checks that all the metrics views referenced by the canvas' components have the same first_day_of_week and first_month_of_year.
+func (r *CanvasReconciler) validateMetricsViewTimeConsistency(ctx context.Context, refs []*runtimev1.ResourceName) error {
+	metricsViews := make(map[string]*runtimev1.Resource)
+	for _, ref := range refs {
+		// Skip non-component refs
+		if ref.Kind != runtime.ResourceKindComponent {
+			continue
+		}
+		component, err := r.C.Get(ctx, ref, false)
+		if err != nil {
+			continue
+		}
+
+		// Skip non-metrics view refs
+		for _, ref := range component.Meta.Refs {
+			if ref.Kind != runtime.ResourceKindMetricsView {
+				continue
+			}
+			// Skip if the metrics view is already in the map
+			if _, ok := metricsViews[ref.Name]; ok {
+				continue
+			}
+			// Get the metrics view
+			mv, err := r.C.Get(ctx, ref, false)
+			if err != nil {
+				if errors.Is(err, drivers.ErrResourceNotFound) {
+					return fmt.Errorf("component %q: metrics view %q not found", ref.Name, ref.Name)
+				}
+				return err
+			}
+
+			// Skip if the metrics view is not valid
+			mvSpec := mv.GetMetricsView().State.ValidSpec
+			if mvSpec == nil {
+				continue
+			}
+
+			metricsViews[ref.Name] = mv
+		}
+	}
+
+	// Validate all metrics views have consistent first_day_of_week or first_month_of_year
+	if len(metricsViews) > 0 {
+		first := false
+		var firstDayOfWeek uint32
+		var firstMonthOfYear uint32
+		var firstViewName string
+
+		for mvName, mv := range metricsViews {
+			mvSpec := mv.GetMetricsView().State.ValidSpec
+			if mvSpec == nil {
+				return status.Errorf(codes.Internal, "metrics view %q in valid spec not found", mvName)
```

Comment:
This comment appears unaddressed here: https://github.com/rilldata/rill/pull/7192/files#r2066263587